substrate-node/pallets/pallet-smart-contract/src/migration.rs

```diff
@@ -38,48 +147,34 @@ pub mod v5 {
     }
 }
 
-pub fn migrate_to_version_6<T: Config>() -> frame_support::weights::Weight {
-    if PalletVersion::<T>::get() == types::StorageVersion::V5 {
+pub fn migrate_to_version_7<T: Config>() -> frame_support::weights::Weight {
+    if PalletVersion::<T>::get() == types::StorageVersion::V6 {
         info!(
             " >>> Starting contract pallet migration, pallet version: {:?}",
             PalletVersion::<T>::get()
         );
 
         let mut migrated_count = 0;
 
-        // Collect ContractsToBillAt storage in memory
-        let contracts_to_bill_at = ContractsToBillAt::<T>::iter().collect::<Vec<_>>();
+        // fix billing information
 
-        // Remove all items under ContractsToBillAt
-        frame_support::migration::remove_storage_prefix(
-            b"SmartContractModule",
-            b"ContractsToBillAt",
-            b"",
-        );
+        translate_node_objects::<T>();
```

Comment:
I don't feel like this code belongs in this file. A migration that translates objects from another pallet should reside in that pallet. Either a global "grid" migration is defined somewhere (not in pallets directory) or we split up migration of objects to the pallets where they belong.

Reply:
Indeed I thought so too. But does the migration happen first in tfgrid pallet?

Reply:
You can order migrations with the executive pallet. So the pallet tfgrid migration can run before the smart contract pallet migration.